packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import getMinutes from 'date-fns/get_minutes';
+import getHours from 'date-fns/get_hours';
+import getDate from 'date-fns/get_date';
+import getMonth from 'date-fns/get_month';
+import getYear from 'date-fns/get_year';
+import setDate from 'date-fns/set_date';
+import setMinutes from 'date-fns/set_minutes';
+import lastDayOfMonth from 'date-fns/last_day_of_month';
+import { twoDigits } from '../pickers/TimePicker';
+import DateTimePicker from '../DateTimePicker';
+import theme from './InputDateTimePicker.scss';
+
+const splitDateAndTimeRegex = new RegExp(/^(.*),(.*)$/);
+const dateRegex = new RegExp(/^\s*([0-9]{1,2})\/([0-9]{1,2})\/([0-9]{4})\s*$/);
+const timeRegex = new RegExp(/^\s*([0-9]{1,2}):([0-9]{2})\s*$/);
+
+function hoursAndMinutesToTime(hours, minutes) {
+	return hours * 60 + minutes;
+}
+
+function getTextDate(date, time) {
+	if (date === undefined) {
+		return '';
+	}
+
+	const day = getDate(date);
+	const month = getMonth(date) + 1;
+	const year = getYear(date);
+
+	const dateText = `${twoDigits(day)}/${twoDigits(month)}/${year}`;
+
+	if (time === undefined) {
+		return dateText;
+	}
+
+	const hours = Math.floor(time / 60);
+	const minutes = time % 60;
+
+	const timeText = `, ${twoDigits(hours)}:${twoDigits(minutes)}`;
+
+	return dateText + timeText;
+}
+
+
+class InputDateTimePicker extends React.Component {
+	static propTypes = {
+		selectedDateTime: PropTypes.instanceOf(Date),
+		onChange: PropTypes.func.isRequired,
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = (() => {
+			const selectedDateTime = this.props.selectedDateTime;
+			if (selectedDateTime !== undefined) {
+				const date = selectedDateTime;
+				const hours = getHours(selectedDateTime);
+				const minutes = getMinutes(selectedDateTime);
+				const time = hoursAndMinutesToTime(hours, minutes);
+				return {
+					date,
+					time,
+					textInput: getTextDate(date, time),
+				};
+			}
+
+			return {
+				textInput: '',
+			};
+		})();
+
+		this.onChangeInput = this.onChangeInput.bind(this);
+		this.onSubmitPicker = this.onSubmitPicker.bind(this);
+	}
+
+	onSubmitPicker({ date, time }) {
+		this.updateDateTime(date, time);
+	}
```

Comment:
Can you try to align your 2 apis to just pass `this.updateDateTime` in the render, and remove this almost useless method ?

Reply:
Nop I cannot :P
Even if it was exactly the same function signature, things always change. Signatures change, code is added/modified, and stick exactly one behavior for one event with the same exact signature will more than likely when changes occurs, give extra params not wanted or less or whatelse not beeing seen easily because it's all implicit.

And in this case for future PR, I've got already added some more behavior when this event occurs, so I definitely don't change it :)

Reply:
I don't understand your argument here. Even with your solution, changing any signature implies a code adaptation.
If you want to be flexible on that, just align all the signatures to take an object so you can add any property you want.

Reply:
If `onSubmit` output a second argument, this will break `this.updateDateTime`.
If we need to do different things on submit, this implies breaking it down to this function, so lots of lines changes in diff for nothing, here, i could just add one behavior/action with just a one more line change.
`this.updateDateTime` is not an event handler anyway and directly passing a 'action' function to an handler lead to errors in present or future because we implicitly pass args between two interfaces that are not the same.

Reply:
✘